Initial Windows version

convey-gilbert opened the pull request:
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A
- [x] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

convey-gilbert committed:
Windows: initial Version

Windows 10 - only for the moment, using an OpenCVBridge lib. The Windows 8.1 version will need the integration of OpenCV via Media Transform Filter.

convey-gilbert committed:
Windows: argscheck

convey-gilbert committed:
Windows: argscheck typo

convey-gilbert committed:
Windows: added file result feature if !returnBase64

convey-gilbert committed:
Windows: some style improvements for camera UI

convey-gilbert commented:
This is a first version supporting Windows 10. Edge detection is based on OpenCV like in the Android version of the plugin. The camera window UI and photo capture code is entirely written in JavaScript using Windows 10 only APIs and works with all Windows 10 versions using the APIs appropriate to the version. If the device supports a separate photo stream, the photo is captured from that, otherwise the video stream is used to capture the photo.
The OpenCVBridge wrapper is written C++. OpenCV 3.1. sourc code is included together with the binaries to be able to rebuild the dependent modules. The elswhere available binary OpenVC DLLs won't confirm to Microsoft Store specifications due to wrong compiler and linker flags used. So it's better to include all here in the project.
There are some additional options to provide to the scanDoc function. Description in detail is following later.
A Windows 8.1 branch of the plugin, using a Media Transfrom Filter instead of a Media Frame Reader is in work and will be provided with coming pull request later, maybe next week.

convey-gilbert committed:
Windows: initial version including Windows 8.1

convey-gilbert committed:
Windows: some bugfixes at versionInfo and property uuids after merge of Win8 code

convey-gilbert committed:
Windows 10: minor optimization at points detection

ChrisTomAlx commented:
Hey @convey-gilbert 
This is brilliant!! Really appreciate your contribution.

Unfortunately though I have no way of testing this, so could you maybe send me a video of your tests? I will then test and make sure this pr doesn't break any of the other platforms (Technically it shouldn't but I'd rather not risk it) and then merge it.

Cheers and have a nice day :)
[Chris](https://github.com/ChrisTomAlx)
[Neutrinos](http://www.neutrinos.co/)

convey-gilbert committed:
Windows: functional version for Windows 8.1 with Media Transform Filter

convey-gilbert committed:
Android: decreased version limit

convey-gilbert committed:
Android: changed build.gradle for backward compatibility

convey-gilbert committed:
Windows: added use of LowLagPhotoCapture API if available on device

convey-gilbert committed:
Windows: strokeWidth of canvas overlay incerased

convey-gilbert committed:
Windows: finished rotationDegree parameter support

convey-gilbert commented:
Hello Chris,
You can find Videos here:
https://secure.convey.de/publish/leadsuccess/img_0353.mov
https://secure.convey.de/publish/leadsuccess/img_0354.mov

convey-gilbert committed:
iOS: Bugfix for iOS13+ to display scanner UIView in fullscreen

New IRLDocumentScanner.framework version added for aspect ratio fix, too.

convey-gilbert committed:
Windows: added missing props file for Windows Phone 8.1 target

convey-gilbert committed:
Windows 8.1: Fixed exception on Win8.1 due to accidentally deleted former changes

convey-gilbert committed:
Update README.md

convey-gilbert committed:
Windows: support of Win8.1 App on Win10 device

convey-gilbert committed:
Fix: check mocule version instead OS version

and fixed take photo with no-clip

convey-gilbert committed:
Windows 10: less flickering of edges on devices with faster cpu

convey-gilbert committed:
Windows 10: minor improvement on auto shutter

convey-gilbert committed:
Windows: deleted accidentally re-inserted version check again

convey-gilbert committed:
Windows 8.1: OpenCV DLL deploy added

convey-gilbert committed:
Windows: optimized recognition smoothing

convey-gilbert committed:
Update documentScannerProxy.js

convey-gilbert committed:
Windows 10: Convert to grayscale added

hung91 committed:
scanlibrary 3.3 version update

hung91 committed:
version reverted to 2.51 (scanlibrary)

hung91 committed:
remove define ordova-plugin-document-scanner.scan

hung91 committed:
test - ignore undefined options

hung91 committed:
revert changes rotationDegree, brightnessValue, focusValue, contrastValue

convey-gilbert committed:
Windows: Added front camera selection

convey-gilbert committed:
Windows: Added sourceType 2 Front Camera. Bugfix  photo capture

prepareLowLagPhotoCaptureAsync with createUncompressed now only for MediaStreamType.photo of type "Video" (for Type "Image" SubType "JPEG" decompression would be needed, for future...)
